### PR TITLE
DOC: IncrementalSearchCV

### PR DESCRIPTION
**Pull request by stsievert**
This PR

* adds in "Attributes" section to `IncrementalSearchCV`
* adds an explanation of `patience`/`tol`
* changes `max_iter=100` by default (mentioned in the docs, but not implemented correctly?)

This PR is based off of #404. A diff between them can be seen at https://github.com/stsievert/dask-ml/pull/1

### CODE REVIEW

**Comment by mrocklin** [comment]
The docs stuff here seems fine to me.  Did you mean to also include the cv_results changes?

**Comment by mrocklin** [comment]
> The docs stuff here seems fine to me. Did you mean to also include the cv_results changes?

Ah, I see as you said above you're basing this off of #404 .  I'll wait until that one goes in before I review this.

**Comment by TomAugspurger** [comment]
Merged master.

File: dask_ml/model_selection/_incremental.py [hunk 1]
```diff
@@ -705,6 +706,72 @@ class IncrementalSearchCV(BaseIncrementalSearchCV):
...
+        * ``model_id``
+        * ``params``
+        * ``partial_fit_calls``
+
+        The key ``model_id`` corresponds to the ``model_id`` in ``cv_results_``.
+        This list of dicts can be imported into Pandas.
+
+    best_score_ : float
+        Best score achieved on the hold out data, where "best" means "the highest
+        score on the hold out set after a model's last partial fit call."
```
Comment (TomAugspurger) [inline dask_ml/model_selection/_incremental.py]: So this isn't quite accurate right now, is it? We could have a case of

Model | Round | Score
----- | ----- | -----
A     | 0     | 0.8
B     | 0     | 0.75
A     | 1     | 0.7
A     | 2     | 0.65

Model A is the chosen as the "best". How about something like the following:

```
best_score_ : float
    The best model's score on the hold-out data in the final round
    of training. The "best" model is the model with the highest
    final score, among all models being trained in the final round.
```

Comment (stsievert) [reply dask_ml/model_selection/_incremental.py]: Maybe 

```
best_score_ : float
    The best model's score on the hold-out data in the final round
    of training.
best_estimator_ : BaseEstimator
    The model with highest final validation score among all the models
    retained by the "inverse decay" algorithm mentioned above.
```

I'm thinking of "above" to be https://github.com/dask/dask-ml/pull/405/files#diff-0a54ded74c8013caf0588e9a5c879e99R628

Comment (stsievert) [reply dask_ml/model_selection/_incremental.py]: Another edit:

```
best_estimator_ : BaseEstimator
    The model with highest final validation score among all the models
    retained by the "inverse decay" algorithm mentioned above.
best_score_ : float
    The score that `best_estimator_` achieved on the validation set
    after the final `partial_fit` call.
```

**Pull request merged**